src/type.cpp

```diff
@@ -1008,7 +1010,7 @@ llvm::Type *PointerType::LLVMType(llvm::LLVMContext *ctx) const {
             if (baseType->IsVoidType())
                 ptype = LLVMTypes::VoidPointerType;
             else
-                ptype = llvm::PointerType::get(baseType->LLVMType(ctx), 0);
+                ptype = llvm::PointerType::get(baseType->LLVMType(ctx, isStorageType), 0);
```

Comment:
I think it needs to be always storage type here, as pointer points to the storage (memory) by definition.